torch/csrc/lazy/core/ir.cpp

```diff
@@ -127,6 +131,20 @@ const Shape& Node::shape(size_t output_index) const {
   return shapes_.at(output_index);
 }
 
+// Add the shape generated by the shape_fn
+void Node::addShapeFromFunction(const std::function<Shape()>& shape_fn) {
```

Comment:
How about naming it 'Node::computeShape'?

and i think the one thing that should be explained in comments is
"although we could simply call shape_fn to compute the node shape, we let the backend interpose so it can cache the computation"

Reply:
> What the method really does is add a function not a shape

I disagree. We aren't adding the function to the node. But rather computing the shape from the function. I liked your suggestion from before and applied that.

Reply:
whoops I was confused the second time then not the first time.

i'm not too clear why you separated addShapeFromFunction from computeShape though.

Reply:
https://github.com/pytorch/pytorch/pull/75324#discussion_r844136060